emu/src/arm/cpu_modes.rs

```diff
@@ -0,0 +1,33 @@
+#[derive(Debug, PartialEq, Eq)]
+pub enum Mode {
+    User = 0b10000,
+    Fiq = 0b10001,
+    Irq = 0b10010,
+    Supervisor = 0b10011,
+    Abort = 0b10111,
+    Undefined = 0b11011,
+    System = 0b11111,
+}
+
+impl From<Mode> for u32 {
+    fn from(m: Mode) -> Self {
+        m as Self
+    }
+}
+
+impl TryFrom<u32> for Mode {
```

Comment:
Why both implementation? (from and tryfrom)

Reply:
One is from `Mode` to `u32` and one try from `u32` to `Mode`.

Do you think they are redundant?

Reply:
For me we can just add what we need, also from and tryfrom are different because for first case you will do `into()` (no failure)... Instead a tryfrom requires a `try_into()` with a possible failure

Reply:
Ok make sense